TRENZ.Extensions.Infisical/InfisicalConfigurationProvider.cs

```diff
@@ -117,16 +117,17 @@ private bool CheckHasChanged(IDictionary<string, SecretElement> newSecrets)
 
     private void LoadSecretsWithTimeout(Action<IDictionary<string, SecretElement>?> callback, TimeSpan timeout)
     {
-        try
-        {
-            var task = Task.Run(() => callback(client.GetAll()));
+        var loadTask = client.GetAllSecretsAsync(CancellationToken.None);
 
-            _ = task.Wait(timeout);
-        }
-        catch (Exception e)
+        if (loadTask.Wait(timeout))
```

Comment:
hm, you shouldn't synchronously wait here (make `LoadSecretsWithTimeout` async as well, I guess?)